scripts/cluster_topology_observer.py

```diff
@@ -29,25 +34,43 @@ def main():
 
     Watch the Patroni API cluster info. When changes are detected, dispatch the change event.
     """
-    patroni_url, run_cmd, unit, charm_dir = sys.argv[1:]
+    patroni_urls, run_cmd, unit, charm_dir = sys.argv[1:]
 
     previous_cluster_topology = {}
+    urls = [urljoin(url, PATRONI_CLUSTER_STATUS_ENDPOINT) for url in patroni_urls.split(",")]
+    member_name = unit.replace("/", "-")
     while True:
         # Disable TLS chain verification
         context = create_default_context()
         context.check_hostname = False
         context.verify_mode = CERT_NONE
 
-        # Scheme is generated by the charm
-        resp = urlopen(  # noqa: S310
-            f"{patroni_url}/{PATRONI_CLUSTER_STATUS_ENDPOINT}",
-            timeout=API_REQUEST_TIMEOUT,
-            context=context,
-        )
-        cluster_status = json.loads(resp.read())
-        current_cluster_topology = {
-            member["name"]: member["role"] for member in cluster_status["members"]
-        }
+        cluster_status = None
+        for url in urls:
```

Comment:
Going forward, we can most probably start and asyncio event loop and call all the units together, instead of going one by one. We should be able to do the same inside the charm as well, where we poll various endpoints.